crates/rust-analyzer/src/main_loop.rs

```diff
@@ -549,14 +555,35 @@ impl GlobalState {
             }
             // Diagnostics are triggered by the user typing
             // so we run them on a latency sensitive thread.
+            let slice2 = slice.clone();
             self.task_pool.handle.spawn(ThreadIntent::LatencySensitive, {
                 let snapshot = self.snapshot();
                 let subscriptions = subscriptions.clone();
```

Comment:
```suggestion
            self.task_pool.handle.spawn(ThreadIntent::LatencySensitive, {
                let snapshot = self.snapshot();
                let subscriptions = subscriptions.clone();
            	let slice = slice.clone();
```
no need to come up with a new name then